pandas/_libs/tslib.pyx

```diff
@@ -44,56 +61,67 @@ from pandas._libs.tslibs.timestamps cimport create_timestamp_from_ts
 from pandas._libs.tslibs.timestamps import Timestamp
 
 from pandas._libs.tslibs.tzconversion cimport (
-    tz_convert_single, tz_convert_utc_to_tzlocal)
+    tz_convert_single,
+    tz_convert_utc_to_tzlocal,
+)
 
 
 cdef inline object create_datetime_from_ts(
-        int64_t value, npy_datetimestruct dts,
-        object tz, object freq):
-    """ convenience routine to construct a datetime.datetime from its parts """
+    int64_t value, npy_datetimestruct dts, object tz, object freq
+):
+    """
+    Convenience routine to construct a datetime.datetime from its parts.
+    """
     return datetime(dts.year, dts.month, dts.day, dts.hour,
                     dts.min, dts.sec, dts.us, tz)
 
 
 cdef inline object create_date_from_ts(
-        int64_t value, npy_datetimestruct dts,
-        object tz, object freq):
-    """ convenience routine to construct a datetime.date from its parts """
+    int64_t value, npy_datetimestruct dts, object tz, object freq
+):
+    """
+    Convenience routine to construct a datetime.date from its parts.
+    """
     return date(dts.year, dts.month, dts.day)
 
 
 cdef inline object create_time_from_ts(
-        int64_t value, npy_datetimestruct dts,
-        object tz, object freq):
-    """ convenience routine to construct a datetime.time from its parts """
+    int64_t value, npy_datetimestruct dts, object tz, object freq
+):
+    """
+    Convenience routine to construct a datetime.time from its parts.
+    """
     return time(dts.hour, dts.min, dts.sec, dts.us, tz)
 
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
-def ints_to_pydatetime(const int64_t[:] arr, object tz=None, object freq=None,
-                       str box="datetime"):
+def ints_to_pydatetime(
+    const int64_t[:] arr,
+    object tz=None,
+    object freq=None,
+    str box="datetime"
+):
     """
     Convert an i8 repr to an ndarray of datetimes, date, time or Timestamp
 
     Parameters
     ----------
-    arr  : array of i8
-    tz   : str, default None
-         convert to this timezone
-    freq : str/Offset, default None
-         freq to convert
-    box  : {'datetime', 'timestamp', 'date', 'time'}, default 'datetime'
-         If datetime, convert to datetime.datetime
-         If date, convert to datetime.date
-         If time, convert to datetime.time
-         If Timestamp, convert to pandas.Timestamp
+    arr : array of i8
+    tz : str, optional
+        convert to this timezone
+    freq : str/Offset, optional
+        freq to convert
+    box : {'datetime', 'timestamp', 'date', 'time'}, default 'datetime'
+        If datetime, convert to datetime.datetime
+        If date, convert to datetime.date
+        If time, convert to datetime.time
+        If Timestamp, convert to pandas.Timestamp
 
     Returns
     -------
-    result : array of dtype specified by box
+    array of dtype specified by box
```

Comment:
dtype is always object, the type contained is specified by box.  let's also specify ndarray instead of just array